krkn/prometheus/client.py

```diff
@@ -251,11 +252,22 @@ def metrics(
                         metric[k] = v
                         metric['timestamp'] = str(datetime.datetime.now())
                     metrics_list.append(metric.copy())
-        if elastic:
+        # Save metrics based on elastic availability
+        if elastic is not None and hasattr(elastic, 'upload_metrics_to_elasticsearch'):
+            # Upload to elasticsearch if elastic client is available
             result = elastic.upload_metrics_to_elasticsearch(
                 run_uuid=run_uuid, index=elastic_metrics_index, raw_data=metrics_list
             )
             if result == -1:
                 logging.error("failed to save metrics on ElasticSearch")
+        else:
+            # Otherwise save locally
+            local_path = os.path.join(tempfile.gettempdir(), f"krkn_metrics_{run_uuid}.json")
```

Comment:
you imported the write_json_to_tmp function but didn't actually use it. Lets replace this code block with using that function